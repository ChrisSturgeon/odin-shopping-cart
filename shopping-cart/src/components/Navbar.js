import { Link } from 'react-router-dom';
import '../styles/navbar.css';

export default function Navbar(props) {
  return (
    <div className="nav-bar">
      <Link className="logo" to="/">
        React Records
      </Link>
      <div className="nav-links">
        <Link to="/">Home</Link>
        <Link to="shop">Shop</Link>
        <Link to="cart">Cart</Link>
        <Link to="about">About</Link>
      </div>
      <div>{props.cart.length} items in cart</div>
    </div>
  );
}
