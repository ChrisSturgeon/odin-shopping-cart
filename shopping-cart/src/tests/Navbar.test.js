import React from 'react';
import { render, screen } from '@testing-library/react';
import Navbar from '../components/Navbar';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

describe('NavBar component', () => {
  it('renders the correct heading', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(screen.getByText('React Records')).toBeInTheDocument();
  });

  it('renders snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

it('has correct home link text', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
  expect(screen.getByTestId('homeLink')).toHaveTextContent('React Records');
});

it('home link has correct class', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
  expect(screen.getByText('React Records')).toHaveClass('active');
});

it('router applies active styling to clicked link', () => {
  render(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>
  );
  const aboutLink = screen.getByRole('link', { name: 'About' });

  userEvent.click(aboutLink);

  expect(aboutLink).toHaveClass('active');
});
