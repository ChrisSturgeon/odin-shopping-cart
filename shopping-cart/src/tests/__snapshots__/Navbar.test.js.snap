// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav-bar"
  >
    <a
      aria-current="page"
      class="logo active"
      data-testid="homeLink"
      href="/"
    >
      React Records
    </a>
    <div
      class="nav-links"
    >
      <a
        class=""
        href="/shop"
      >
        Shop
      </a>
      <a
        class=""
        href="/about"
      >
        About
      </a>
      <a
        class=""
        href="/contact"
      >
        Contact
      </a>
    </div>
  </div>
</DocumentFragment>
`;
